opensearch-transport/lib/opensearch/transport/transport/base.rb

```diff
@@ -424,14 +433,14 @@ def apply_headers(client, options)
         end
 
         def find_value(hash, regex)
-          key_value = hash.find { |k,v| k.to_s.downcase =~ regex }
+          key_value = hash.find { |k, _v| k.to_s.downcase =~ regex }
           if key_value
             hash.delete(key_value[0])
             key_value[1]
           end
```

Comment:
(Not relevant to this PR but just a fun thought)
This could be a one-liner :P
`hash.delete (hash.keys.find { |k| k.to_s.downcase =~ regex })`

Because `Hash#delete` will also return the value if the key is present, else `nil` is returned.

Reply:
I think a good next step is to actually go through the code, fix rubocop violations instead of ignoring them, and improve things like these!